### PR TITLE
Add API stubs for the WebUSB transfer result interfaces

### PR DESCRIPTION
**Pull request by reillyeon**
Please check to make sure I didn't make any copy-paste errors.

### CODE REVIEW

File: content/en-US/api/USBInTransferResult/USBInTransferResult.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+## Constructor
+
+Creates a new `USBInTransferResult` object with the provided `status` and `data` fields.
+
+## Properties
+
+**`USBInTransferResult.data`**
+
+Returns a `DataView` over the data returned from the USB device, if any.
```
Comment (jpmedley) [inline content/en-US/api/USBInTransferResult/USBInTransferResult.md]:
```suggestion
Returns a `DataView` object containing the data returned from the USB device, if any.
```

File: content/en-US/api/USBInTransferResult/USBInTransferResult.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+**`USBInTransferResult.status`**
+
+Returns the status of the transfer request, one of:
+
+* `"ok"` - The transfer was successful.
+* `"stall"` - The device indicated an error by generating a stall condition on the endpoint. A stall on the control endpoint does not need to be cleared. For other pipes a call to `clearHalt()` is required before `transferIn()` can be called again.
```
Comment (jpmedley) [inline content/en-US/api/USBInTransferResult/USBInTransferResult.md]: 'For other pipes...'

How does this relate to `"stall"`?

File: content/en-US/api/USBIsochronousInTransferPacket/USBIsochronousInTransferPacket.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+## Constructor
+
+Creates a new `USBIsochronousInTransferPacket` object with the provided `status` and `data` fields.
+
+## Properties
+
+**`USBIsochronousInTransferPacket.data`**
+
+Returns a `DataView` over the data returned from the USB device in this packet, if any.
```
Comment (jpmedley) [inline content/en-US/api/USBIsochronousInTransferPacket/USBIsochronousInTransferPacket.md]:
```suggestion
Returns a `DataView` object containing the data returned from the USB device in this packet, if any.
```

File: content/en-US/api/USBIsochronousInTransferResult/USBIsochronousInTransferResult.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+## Constructor
+
+Creates a new `USBIsochronousInTransferResult` object with the provided `packets` and `data` fields.
+
+## Properties
+
+**`USBIsochronousInTransferResult.data`**
+
+Returns a `DataView` over the buffer into which data from the device was written. This is the combined data from all packets. See the individual `DataView`s in the `packets` array for the portion of this buffer containing data from each packet.
```
Comment (jpmedley) [inline content/en-US/api/USBIsochronousInTransferResult/USBIsochronousInTransferResult.md]:
```suggestion
Returns a `DataView` object containing the buffer into which data from the device was written. This is the combined data from all packets. See the individual `DataView` objects in the `packets` array for the portion of this buffer containing data from each packet.
```

File: content/en-US/api/USBIsochronousOutTransferResult/USBIsochronousOutTransferResult.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+## Constructor
+
+Creates a new `USBIsochronousOutTransferResult` object with the provided `packet` field.
+
+## Properties
+
+**`USBIsochronousOutTransferResult.packets`**
+
+Returns an array of `USBIsochronousOutTransferPacket`s containing the result of each request to send a packet to the device.
```
Comment (jpmedley) [inline content/en-US/api/USBIsochronousOutTransferResult/USBIsochronousOutTransferResult.md]:
```suggestion
Returns an array of `USBIsochronousOutTransferPacket` objects containing the result of each request to send a packet to the device.
```

File: content/en-US/api/USBOutTransferResult/USBOutTransferResult.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+**`USBOutTransferResult.bytesWritten`**
+
+Returns the number of bytes from the transfer request that were sent to the device.
+
+**`USBOutTransferResult.status`**
+
+Returns the status of the transfer request, one of:
+
+* `"ok"` - The transfer was successful.
+* `"stall"` - The device indicated an error by generating a stall condition on the endpoint. A stall on the control endpoint does not need to be cleared. For other pipes a call to `clearHalt()` is required before `transferOut()` can be called again.
```
Comment (jpmedley) [inline content/en-US/api/USBOutTransferResult/USBOutTransferResult.md]: 'For other pipes...' Same question here as above.

Does this property als return `"babble"`?
```suggestion
* `"stall"` - The device indicated an error by generating a stall condition on the endpoint. A stall on the control endpoint does not need to be cleared. For other pipes a call to `clearHalt()` is required before `transferOut()` can be called again.
```

File: content/en-US/api/USBIsochronousOutTransferPacket/USBIsochronousOutTransferPacket.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+**`USBIsochronousOutTransferPacket.bytesWritten`**
+
+Returns the number of bytes from the packet that were sent to the device.
+
+**`USBIsochronousOutTransferPacket.status`**
+
+Returns the status of the transfer request, one of:
+
+* `"ok"` - The transfer was successful.
+* `"stall"` - The device indicated an error by generating a stall condition on the endpoint. A stall on an isochronous endpoint does not need to be cleared.
```
Comment (jpmedley) [inline content/en-US/api/USBIsochronousOutTransferPacket/USBIsochronousOutTransferPacket.md:28]: Does this property return `"babble"`?

**Review by jpmedley — CHANGES_REQUESTED** [review]
Reilly,

Thanks for the submission. I have some question, when you get a moment.

Also, some of the requested changes are edits that I'm asking you to merge. They bring the text more in line with what's done on MDN.

Thanks,
Joe

File: content/en-US/api/USBOutTransferResult/USBOutTransferResult.md (hunk 6 above)
Comment (reillyeon) [reply content/en-US/api/USBOutTransferResult/USBOutTransferResult.md]: No, babble is not possible for an out endpoint.

File: content/en-US/api/USBInTransferResult/USBInTransferResult.md (hunk 2 above)
Comment (reillyeon) [reply content/en-US/api/USBInTransferResult/USBInTransferResult.md]: I've clarified the description a bit. What do you think?

**Pull request merged**